core/src/com/unciv/UncivGame.kt

```diff
@@ -212,67 +221,84 @@ class UncivGame(parameters: UncivGameParameters) : Game() {
-        setScreen(screen)
+        throw UnsupportedOperationException("Use pushScreen or replaceCurrentScreen instead")
     }
 
     override fun getScreen(): BaseScreen? {
         val curScreen = super.getScreen()
         return if (curScreen == null) { null } else { curScreen as BaseScreen }
     }
 
-    /** Sets the screen of the game and automatically [disposes][Screen.dispose] the old screen as long as it isn't the world screen. */
-    fun setScreen(newScreen: BaseScreen) {
+    private fun setScreen(newScreen: BaseScreen) {
+        debug("Setting new screen: %s, screenStack: %s", newScreen, screenStack)
+        Gdx.input.inputProcessor = newScreen.stage
+        super.setScreen(newScreen) // This can set the screen to the policy picker or tech picker screen, so the input processor must be set before
         if (newScreen is WorldScreen) {
-            debug(
-                "Setting new world screen: gameId: %s, turn: %s, curCiv: %s",
-                newScreen.gameInfo.gameId, newScreen.gameInfo.turns, newScreen.gameInfo.currentPlayer
-            )
-            if (newScreen != worldScreen) worldScreen?.dispose()
-            worldScreen = newScreen
             newScreen.shouldUpdate = true
-            Gdx.graphics.requestRendering()
-        } else {
-            debug("Setting new screen: %s", newScreen)
         }
+        Gdx.graphics.requestRendering()
+    }
 
-        val oldScreen = screen
-        Gdx.input.inputProcessor = newScreen.stage
-        super.setScreen(newScreen) // This can set the screen to the policy picker or tech picker screen, so the input processor must be set before
-        if (oldScreen !is WorldScreen) { // we want to keep the world screen around, because it's expensive to re-create it
-            oldScreen?.dispose()
-        }
+    /** Adds a screen to be displayed instead of the current screen, with an option to go back to the previous screen by calling [popScreen] */
+    fun pushScreen(newScreen: BaseScreen) {
+        screenStack.addLast(newScreen)
+        setScreen(newScreen)
     }
 
     /**
-     * Resets the game to the stored world screen and automatically [disposes][Screen.dispose] the old screen.
+     * Pops the currently displayed screen off the screen stack and shows the previous screen. If there is no previous screen, will ask the user to quit the game.
+     * Automatically [disposes][BaseScreen.dispose] the old screen as long as it isn't the world screen.
+     *
+     * @return the new screen
      */
-    fun resetToWorldScreen() {
-        setScreen(worldScreen!!)
+    fun popScreen(): BaseScreen? {
+        val oldScreen = screenStack.removeLast()
+        val newScreen = screenStack.lastOrNull()
+        if (newScreen == null) {
+            return null
+        }
+        setScreen(newScreen)
+        if (oldScreen !is WorldScreen) { // we want to keep the world screen around, because it's expensive to re-create it
```

Comment:
Sounds like a dangerous pitfall requiring outside code to pay attention. Who would be keeping a viable reference and be able to reuse it? Sounds like that expertise may be better encapsulated close to these routines here, maybe even do a dedicated BaseScreenStack class...